src/Features/ExternalAccess/Copilot/Internal/ContextProviders/CSharpCopilotMethodCallersService.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.ExternalAccess.Copilot.ContextProviders;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Copilot.Internal.ContextProviders;
+
+[ExportLanguageService(typeof(ICopilotMethodCallersService), language: LanguageNames.CSharp), Shared]
+internal class CSharpCopilotMethodCallersService : ICopilotMethodCallersService
+{
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public CSharpCopilotMethodCallersService()
+    {
+    }
+
+    public async Task<ISymbol?> GetContainingMethodSymbolAsync(Document document, int position, CancellationToken cancellationToken)
+    {
+        var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+        if (semanticModel is null)
+        {
+            return null;
+        }
+
+        return semanticModel.GetEnclosingSymbol<IMethodSymbol>(position, cancellationToken);
+    }
+
+    public async Task GetMethodCallersAsync(Document document, ISymbol symbol, Func<Document, TextSpan, CancellationToken, ValueTask> reporter, CancellationToken cancellationToken)
```

Comment:
If this is 'get method callers' this should take an IMethodSymbol. 

Why is this passed a lambda, instead of just being a call to find refs by the caller?